File: leetcode/2946-matrix-similarity-after-cyclic-shifts/2946-matrix-similarity-after-cyclic-shifts.java
class Solution {
    public boolean areSimilar(int[][] mat, int k) {
        int m = mat.length;
        int n = mat[0].length;

        int shift = k % n; // 유효한 이동 거리

        if (shift == 0) {
            return true;
        }

        for (int r = 0; r < m; r++) {
            for (int c = 0; c < n; c++) {

                if (r % 2 == 0) {
                    // 짝수 행: 왼쪽으로 shift
                    int nc = (c + shift) % n;
                    if (mat[r][c] != mat[r][nc]) {
                        return false;
                    }
                } else {
                    // 홀수 행: 오른쪽으로 shift
                    int nc = (c - shift + n) % n;
                    if (mat[r][c] != mat[r][nc]) {
                        return false;
                    }
                }
            }
        }
        return true;
    }
}

File: leetcode/2946-matrix-similarity-after-cyclic-shifts/2946-matrix-similarity-after-cyclic-shifts.js
function areSimilar(mat, k) {
  const m = mat.length;
  const n = mat[0].length;
  const shift = k % n; // effective shift

  if (shift === 0) return true; // whole matrix returns to original

  for (let r = 0; r < m; r++) {
    for (let c = 0; c < n; c++) {
      if (r % 2 === 0) {
        // even row: left shift
        const nc = (c + shift) % n;
        if (mat[r][c] !== mat[r][nc]) return false;
      } else {
        // odd row: right shift
        const nc = (c - shift + n) % n;
        if (mat[r][c] !== mat[r][nc]) return false;
      }
    }
  }
  return true;
}